src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineMethod;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.InlineMethod)), Shared]
+    [Export(typeof(CSharpInlineMethodRefactoringProvider))]
+    internal sealed class CSharpInlineMethodRefactoringProvider :
+        AbstractInlineMethodRefactoringProvider<InvocationExpressionSyntax, ExpressionSyntax, ArgumentSyntax, MethodDeclarationSyntax, IdentifierNameSyntax>
+    {
+        /// <summary>
+        /// All the syntax kind considered as the statement contains the invocation callee.
```

Comment:
how is this difference from just finding the ancestor that is a StatementSyntax?

Reply:
that's is not guaranteed to be StatementSyntax. 
Like `if (Callee()) { }` it will be the if statement

Reply:
And that's why I have a list of syntax kind needs to be considered as that [here](https://github.com/Cosifne/roslyn/blob/cf11323dd4a2cb53cc43b7aaf78fa44cc3f4b6ec/src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs#L28)

Reply:
an IfStatemnetSyntax *is* a StatementSyntax.